discovery-infra/oc_login.py

```diff
@@ -0,0 +1,74 @@
+import os
+import yaml
+
+from logger import log
+from utils import run_command
+
+
+def is_oc_login_required(target):
+    return target in ('oc', 'oc-ingress')
+
+
+def oc_login(token=None, server=None):
+    log.info('Performing oc-login')
+    cmd = _get_cmd(token, server)
+    run_command(cmd)
+
+
+def _get_cmd(token, server):
+    if not token or not server:
+        token, server = _find_token_and_server_from_kubeconfig()
+
+    return _build_cmd_from_token_and_server(token, server)
+
+
+def _find_token_and_server_from_kubeconfig():
+    config = _load_kubeconfig()
+    for cluster in _iterate_clusters(config):
+        user = _get_user_by_cluster(config, cluster)
+        if not user:
+            continue
+        token = _get_token_by_user(config, user)
+        if not token:
+            continue
+        server = cluster['cluster']['server']
+        return token, server
+
+    raise RuntimeError(
+        'unable to find any valid pair of token and server in kubeconfig file '
+        'to perform oc-login'
+    )
+
+
+def _load_kubeconfig():
```

Comment:
regarding kubeconfig, why we think it will be default one? if i want to connect to psi i will download it's kubeconfig right? and why i will set it as default one?

Reply:
i think that we must provide path to kubeconfig